source/common/network/apple_dns_impl.cc

```diff
@@ -414,14 +414,16 @@ AppleDnsResolverImpl::PendingResolution::buildDnsResponse(const struct sockaddr*
     address_in.sin_family = AF_INET;
     address_in.sin_port = 0;
     address_in.sin_addr = reinterpret_cast<const sockaddr_in*>(address)->sin_addr;
-    return {std::make_shared<const Address::Ipv4Instance>(&address_in), std::chrono::seconds(ttl)};
+    return {Address::InstanceFactory::createInstancePtrOrThrow<Address::Ipv4Instance>(&address_in),
```

Comment:
Just want to sanity check here that we think this is only called on the main thread. I know that you have the assert in createInstancePtrOrThrow but wasn't 100% sure if the macOS CI runs with asserts.